src/menu/MenuSongSelection.as

```diff
@@ -603,425 +626,1029 @@ package menu
                 yOffset += sI.height + 2;
             }
 
-            // Scroll to last song
-            if (_gvars.tempFlags['scroll_dist_temp'] != null)
-            {
-                var dist:Number = _gvars.tempFlags['scroll_dist_temp'];
-                pane.scrollTo(dist, false);
-                scrollbar.scrollTo(dist, false);
-            }
-            // Init scroll
-            else
-            {
-                pane.scrollTo(scrollbar.scroll, false);
-            }
+            // Scroll Position
+            pane.scrollTo(options.scroll_position, false);
+            scrollbar.scrollTo(options.scroll_position, false);
 
             scrollbar.draggerVisibility = (yOffset > pane.height);
 
-            // Set Active ID if it's not set
-            if (options.activeSongID == -1)
+            //- Update Selected Index
+            // Find and select last active song id.
+            var hasSelected:Boolean = false;
+            for (sX = 0; sX < songList.length; sX++)
             {
-                setActiveID(0, false);
+                song = songList[sX];
+                if (options.activeSongID == song.level)
+                {
+                    setActiveIndex(sX, -1, false);
+                    hasSelected = true;
+                    break;
+                }
             }
-        }
 
-        public function drawPages():void
-        {
-            // Remove pages
-            if (pages != null)
+            // No active valid song found, clear saved actives.
+            if (!hasSelected)
             {
-                this.removeChild(pages);
-                pages = null;
+                options.activeIndex = -1;
+                options.activeSongID = -1;
             }
 
-            pages = new Sprite();
-            pages.y = 424;
+            // No song selected, select the first in the list if valid.
+            if (songItems.length > 0 && options.activeIndex == -1)
```

Comment:
```suggestion
            if (options.activeIndex == -1)
```
Since we have already done a check before this to see whether `songList` is `null` or empty, we could probably simplify this check here.

Reply:
Will change.